File: 스택_큐/주식가격/주식가격.js
function solution(prices) {
    return prices.map((price, i, origin) => {
        if (i === origin.length - 1) {
            return 0
        }
        for (let j = i + 1; j < origin.length; j++) {
            if (price > origin[j]) {
                return j - i
            }
        }
        return origin.length - i - 1
    })
}

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
	public int[] solution(int[] prices) {
		List<Integer> result = new ArrayList<>();
		for (int i = 0; i < prices.length; i++) {
			int time = 0;
			for (int j = i + 1; j < prices.length; j++) {
                time++;
				if (prices[i] > prices[j]) {
					break;
				}
			}
			result.add(time);
		}
		return result.stream().mapToInt(Integer::intValue).toArray();
	}
}
